### PR TITLE
mlm conversion & tiktokenizer support

### PR DESCRIPTION
**Pull request by dimapihtar**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by mikolajblaz — COMMENTED** [review]
Why is the `load_mlm` flag needed? Has something changed in the model construction which results in the absence of the `model.` prefix?

File: nemo/collections/nlp/models/nlp_model.py [hunk 1]
```diff
@@ -397,7 +397,21 @@ def dummy():
...
-                checkpoint['state_dict'] = sharded_state_dict
+                if kwargs.get("load_mlm", False):
+                    mlm_sharded_state_dict = {}
+                    for k, v in sharded_state_dict.items():
+                        # Remove 'model.' from the sharded_state_dict keys
+                        new_key = k.replace('model.', '', 1)
+
+                        # Update the key attribute of the ShardedTensor value
+                        new_value = v
+                        new_value.key = v.key.replace('model.', '', 1) if hasattr(v, 'key') else v.key
```
[the remark this answers is not in the document]
Comment (dimapihtar) [reply nemo/collections/nlp/models/nlp_model.py:408]: @mikolajblaz yes, when I'm trying to load mlm ckpt as nemo ckpt without removing that "model." suffix It gives me KeyError on that step because it's trying to find keys in the checkpoint with "model." suffix.

and the structure looks like this:
{`model.decoder.somelayername`: ShardedTensor(key='model.decoder.somelayername', tensor=tensorvalues)}

so we also want to remove it for tensor keys

**Comment by dimapihtar** [comment]
> approving after the 1 comment above is resolved (I don't seem to have permission to actually approve the PR)

fixed.

File: .github/workflows/cicd-main.yml [hunk 2]
```diff
@@ -2109,6 +2109,126 @@ jobs:
...
+          --nemo_file_path /tmp/mlm_to_nemo_test.nemo \
+          --tensor_model_parallel_size 1 \
+          --pipeline_model_parallel_size 1 \
+          --gpus_per_node 1 \
+          --model_type gpt \
+          --hparams_file /tmp/nemo_conversion_ckpt/megatron_gpt/version_0/hparams.yaml \
+          --convert_mlm
+
+      AFTER_SCRIPT: |
+        rm -rf /tmp/nemo_conversion_ckpt
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:2228]: This is not needed , can be removed

**Review by pablo-garay — APPROVED** [review]
